tools/rtpsrelay/RelayHandler.cpp

```diff
@@ -972,18 +969,14 @@ namespace {
 
         if (submessage_header.flags & OpenDDS::RTPS::FLAG_D) {
           OpenDDS::RTPS::ParameterList plist;
-
-          message_parser.serializer().swap_bytes(!ACE_CDR_BYTE_ORDER); // read "encap" itself in LE
-          CORBA::UShort encap, options;
-          if (!(message_parser >> encap) || (encap != encap_LE && encap != encap_BE)) {
+          OpenDDS::DCPS::EncapsulationHeader encap;
+          if (!(message_parser >> encap) || (encap.kind() != OpenDDS::DCPS::EncapsulationHeader::KIND_PL_CDR_LE &&
+                                             encap.kind() != OpenDDS::DCPS::EncapsulationHeader::KIND_PL_CDR_BE)) {
             ACE_ERROR((LM_ERROR,
                        ACE_TEXT("(%P|%t) ERROR: extract_common_name() - ")
                        ACE_TEXT("failed to deserialize encapsulation header for SPDP\n")));
             return "";
           }
```

Comment:
Same issue as the code in Spdp that needed to set the encoding on the serializer.

Reply:
This still needs the same sort of change as [this comment](https://github.com/objectcomputing/OpenDDS/pull/3044#discussion_r718819742) because it was copied from there.

Reply:
Once it is though, I would consider the PR done.